server/test/repository/ApplicationStatusesRepositoryTest.java

```diff
@@ -0,0 +1,87 @@
+package repository;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatExceptionOfType;
+
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.Random;
+import models.ApplicationStatusesModel;
+import models.ProgramModel;
+import models.StatusDefinitionsLifecycleStage;
+import org.junit.Before;
+import org.junit.Test;
+import services.LocalizedStrings;
+import services.program.StatusDefinitions;
+import support.ProgramBuilder;
+
+public class ApplicationStatusesRepositoryTest extends ResetPostgres {
+  private static final StatusDefinitions.Status APPROVED_STATUS =
+      StatusDefinitions.Status.builder()
+          .setStatusText("Approved")
+          .setLocalizedStatusText(LocalizedStrings.withDefaultValue("Approved"))
+          .build();
+  ApplicationStatusesRepository repo;
+
+  @Before
+  public void setupStatusRepository() {
+    repo = instanceOf(ApplicationStatusesRepository.class);
+  }
+
+  @Test
+  public void canQueryForActiveApplicationStatuses() {
+    // setup
+    Long uniqueProgramId = new Random().nextLong();
+    ProgramModel program =
+        ProgramBuilder.newActiveProgram("Active status tests" + uniqueProgramId, "description")
+            .build();
+    String programName = program.getProgramDefinition().adminName();
+    StatusDefinitions statusDefinitions = new StatusDefinitions(ImmutableList.of(APPROVED_STATUS));
+    // test
+    ApplicationStatusesModel applicationStatusesModel =
+        new ApplicationStatusesModel(
+            programName, statusDefinitions, StatusDefinitionsLifecycleStage.ACTIVE);
+    applicationStatusesModel.save();
+    applicationStatusesModel.setCreateTimeForTest("2041-01-01T00:00:00Z").save();
```

Comment:
**info, non-blocking**: This is a matter of taste but since I mentioned arrange/act/assert before I figured I'd mention:
imo this is also part of the "arrange" step.  the only "act" step is the line calling the code under test, which is `repo.lookupActiveStatusDefinitions(programName)`

Reply:
Done!